### PR TITLE
use latest version of capx in template-tests

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/nutanix-cloud-native/cluster-api-provider-nutanix
 
-go 1.22.1
+go 1.22.2
```
Comment (jimmidyson) [inline go.mod:3]: I think this should stick at 1.21 to be usable by projects at that golang level. If you want to use go 1.22 for builds still you can specify the toolchain to use to build the project specifically. https://go.dev/doc/toolchain is a good resource to understand this. I haven't done this previously myself though but i think this should work...

```suggestion
go 1.21

toolchain go1.22.2
```

**Review by thunderboltsid — APPROVED** [review]
/lgtm
/approve

**Comment by deepakm-ntnx** [comment]
working on following issue <pre>
Cluster Class Template Patches Test Suite patches for failure domains Control Plane NutanixMachineTemplate should not have the cluster and subnets set
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:312
  [FAILED] in [It] - /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 @ 04/11/24 14:45:52.501
• [FAILED] [60.016 seconds]
Cluster Class Template Patches Test Suite patches for failure domains [It] Control Plane NutanixMachineTemplate should not have the cluster and subnets set
/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:312

  [FAILED] Timed out after 60.016s.
  The function passed to Eventually returned the following error:
      <*errors.errorString | 0xc0015133d0>: 
      no control plane NutanixMachineTemplate found for cluster cluster-with-failure-domains
      {
          s: "no control plane NutanixMachineTemplate found for cluster cluster-with-failure-domains",
      }
  In [It] at: /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 @ 04/11/24 14:45:52.501

  Full Stack Trace
    github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates.init.func1.1.3()
    	/Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 +0x1e8
</pre>

**Comment by deepakm-ntnx** [comment]
> working on following issue
> 
> 
> 
> Cluster Class Template Patches Test Suite patches for failure domains Control Plane NutanixMachineTemplate should not have the cluster and subnets set
> 
> /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:312
> 
> [FAILED] in [It] - /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 @ 04/11/24 14:45:52.501
> 
> • [FAILED] [60.016 seconds]
> 
> Cluster Class Template Patches Test Suite patches for failure domains [It] Control Plane NutanixMachineTemplate should not have the cluster and subnets set
> 
> /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:312
> 
> [FAILED] Timed out after 60.016s.
> The function passed to Eventually returned the following error:
> <*errors.errorString | 0xc0015133d0>:
> no control plane NutanixMachineTemplate found for cluster cluster-with-failure-domains
> {
> s: "no control plane NutanixMachineTemplate found for cluster cluster-with-failure-domains",
> }
> In [It] at: /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 @ 04/11/24 14:45:52.501
> 
> 
> Full Stack Trace
> github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates.init.func1.1.3()
> /Users/deepak.muley/go/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix/templates/template_test.go:315 +0x1e8

Seems like this problem is not showing up here again

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/nutanix-cloud-native/cluster-api-provider-nutanix
 
 go 1.22.1
 
+toolchain go1.22.2
```
Comment (deepakm-ntnx) [inline go.mod:5]: Note: devbox did not have 1.22.2 hence had to make it this way

Comment (dkoshkin) [reply go.mod:5]: Why does the version needs to be bumped?

Comment (deepakm-ntnx) [reply go.mod:5]: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/512#pullrequestreview-2001724387
else caren's govulnerability check fails due to http/net version

Comment (jimmidyson) [reply go.mod:5]: This toolchain version wouldn't affect CAREN afaik as it is local to working on the module here.

**Review by jimmidyson — APPROVED** [review]

**Comment by jimmidyson** [comment]
I'm confused by the tide check message: `Not mergeable. Merge type "merge" disallowed by repo settings`

**Review by dkoshkin — APPROVED** [review]

**Comment by thunderboltsid** [comment]
> I'm confused by the tide check message: `Not mergeable. Merge type "merge" disallowed by repo settings`

We have explicitly disabled everything but squash and merge to avoid polluting history with unclean merges and enable easy revert of entire PRs, and I'm guessing it's just complaining because the check is expecting a non-squashed merge. We need to revisit the configuration of that tide check. cc @adiantum

**Pull request merged**